packages/machinomy/src/PaymentValidation.ts

```diff
@@ -60,6 +61,14 @@ export default class PaymentValidation {
     return isValidPaymentValue
   }
 
+  private async isValidPaymentPrice (): Promise<boolean> {
+    const isValidPaymentPrice = this.paymentChannel.spent.add(this.payment.price).eq(this.payment.value)
```

Comment:
@kincaidoneil Here "eq" must be "lessOrEq" for cases when database is accidentally cleared (when "spent" from db become 0) but payments still coming. We experienced this problem here (with old Machinomy code) 1358ae8a0058a3edd05cdad1c40590eeff7ebf03 , two months ago we not quite correctly handled this use case (removed this check due to issues with some Machinomy-related libraries), thank you for finding this bug and contributing a fix!